packages/mgt-components/src/graph/graph.files.ts

```diff
@@ -201,3 +201,427 @@ export async function getUserInsightsDriveItemById(
   } catch {}
   return response || null;
 }
+
+// GET /me/drive/root/children
+export async function getFilesIterator(graph: IGraph, top?: number): Promise<GraphPageIterator<DriveItem>> {
+  const scopes = 'files.read';
+  let request;
+  let filesPageIterator;
+
+  try {
+    if (top) {
+      request = await graph
+        .api('/me/drive/root/children')
+        .top(top)
+        .middlewareOptions(prepScopes(scopes));
+    } else {
+      request = graph.api('/me/drive/root/children').middlewareOptions(prepScopes(scopes));
+    }
+
+    filesPageIterator = await getFilesPageIterator(graph, request);
+  } catch {}
+  return filesPageIterator || null;
+}
+
+// GET /drives/{drive-id}/items/{item-id}/children
+export async function getDriveFilesByIdIterator(
+  graph: IGraph,
+  driveId: string,
+  itemId: string,
+  top?: number
+): Promise<GraphPageIterator<DriveItem>> {
+  const scopes = 'files.read';
+  let request;
+  let filesPageIterator;
+
+  try {
+    if (top) {
+      request = graph
+        .api(`/drives/${driveId}/items/${itemId}/children`)
+        .top(top)
+        .middlewareOptions(prepScopes(scopes));
+    } else {
+      request = graph.api(`/drives/${driveId}/items/${itemId}/children`).middlewareOptions(prepScopes(scopes));
+    }
+    filesPageIterator = await getFilesPageIterator(graph, request);
+  } catch {}
+  return filesPageIterator || null;
+}
+
+// GET /drives/{drive-id}/root:/{item-path}:/children
+export async function getDriveFilesByPathIterator(
+  graph: IGraph,
+  driveId: string,
+  itemPath: string,
+  top?: number
+): Promise<GraphPageIterator<DriveItem>> {
+  const scopes = 'files.read';
+  let request;
+  let filesPageIterator;
+
+  try {
+    if (top) {
+      request = graph
+        .api(`/drives/${driveId}/root:/${itemPath}:/children`)
+        .top(top)
+        .middlewareOptions(prepScopes(scopes));
+    } else {
+      request = graph.api(`/drives/${driveId}/root:/${itemPath}:/children`).middlewareOptions(prepScopes(scopes));
+    }
```

Comment:
You can simplify this here (and everywhere else in this file where you use this pattern), so you don't repeat the same query twice

```suggestion
    request = graph.api(`/drives/${driveId}/root:/${itemPath}:/children`).middlewareOptions(prepScopes(scopes));
    if (top) {
      request.top(top);
    }
```

Reply:
I can update this!